lib/sidekiq/form526_backup_submission_process/processor.rb

```diff
@@ -442,7 +442,7 @@ def choose_provider(headers, provider, breakered: true)
           provider:,
           # this sends the auth headers and if we want the "breakered" or "non-breakered" version
           options: { auth_headers: headers, breakered: },
-          current_user: OpenStruct.new({ flipper_id: submission.user_uuid, icn: @user_account.icn }),
+          current_user: OpenStruct.new({ flipper_id: submission.user_uuid, icn: submission.user_account.icn }),
```

Comment:
Should we add a couple of safe navigation operators here to make the code more robust against nil values at any point in the chain? `submission&.user_account&.icn`

Reply:
Good call. I changed the approach entirely to use the way the primary path sets the user account.